tests/catch/unit/memory/hipMalloc3DArray.cc

```diff
@@ -48,148 +49,403 @@ static constexpr auto ARRAY_LOOP{100};
+
+  hipArray_t array;
+
+#if HT_NVIDIA
+  hipError_t expectedError = hipErrorUnknown;
+#else
+  hipError_t expectedError = hipErrorInvalidValue;
+#endif
+
+  HIP_CHECK_ERROR(hipMalloc3DArray(&array, &desc, makeExtent(flag, s), flag), expectedError);
+}
+
+TEST_CASE("Unit_hipMalloc3DArray_Negative_8BitFloat") {
+  constexpr size_t s = 6;  // 6 to keep cubemap happy
+  hipArray_t array;
+  hipChannelFormatDesc desc = GENERATE(hipCreateChannelDesc(8, 0, 0, 0, hipChannelFormatKindFloat),
+                                       hipCreateChannelDesc(8, 8, 0, 0, hipChannelFormatKindFloat),
+                                       hipCreateChannelDesc(8, 8, 8, 8, hipChannelFormatKindFloat));
+
+  const auto flag = GENERATE(from_range(std::begin(validFlags), std::end(validFlags)));
+
+#if HT_NVIDIA
+  hipError_t expectedError = hipErrorUnknown;
+#else
+  hipError_t expectedError = hipErrorInvalidValue;
+#endif
+
+  HIP_CHECK_ERROR(hipMalloc3DArray(&array, &desc, makeExtent(flag, s), flag), expectedError);
+}
+
+TEST_CASE("Unit_hipMalloc3DArray_Negative_DifferentChannelSizes") {
+#if HT_AMD
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-129");
+  return;
+#endif
+
+  const int bitsX = GENERATE(8, 16, 32);
+  const int bitsY = GENERATE(8, 16, 32);
+  const int bitsZ = GENERATE(8, 16, 32);
+  const int bitsW = GENERATE(8, 16, 32);
+  if (bitsX == bitsY && bitsY == bitsZ && bitsZ == bitsW) return;  // skip when they are equal
+
+  const hipChannelFormatKind channelFormat =
+      GENERATE(hipChannelFormatKindSigned, hipChannelFormatKindUnsigned, hipChannelFormatKindFloat);
+
+  if (channelFormat == hipChannelFormatKindFloat &&
+      (bitsX == 8 || bitsY == 8 || bitsZ == 8 || bitsW == 8))
+    return;  // 8 bit floats aren't allowed
+
+  hipChannelFormatDesc desc = hipCreateChannelDesc(bitsX, bitsY, bitsZ, bitsW, channelFormat);
+
+  INFO("format: " << channelFormatString(channelFormat) << ", x bits: " << bitsX
+                  << ", y bits: " << bitsY << ", z bits: " << bitsZ << ", w bits: " << bitsW);
+
+  const auto flag = GENERATE(from_range(std::begin(validFlags), std::end(validFlags)));
+
+  constexpr size_t s = 6;  // 6 to keep cubemap happy
+  hipArray_t array{};
+
+#if HT_NVIDIA
+  hipError_t expectedError = hipErrorUnknown;
+#else
+  hipError_t expectedError = hipErrorInvalidValue;
+#endif
+
+  HIP_CHECK_ERROR(hipMalloc3DArray(&array, &desc, makeExtent(flag, s), flag), expectedError);
+}
+
+TEST_CASE("Unit_hipMalloc3DArray_Negative_BadChannelSize") {
+  const int badBits = GENERATE(-1, 0, 10, 100);
+  const hipChannelFormatKind formatKind =
+      GENERATE(hipChannelFormatKindSigned, hipChannelFormatKindUnsigned, hipChannelFormatKindFloat);
+  hipChannelFormatDesc desc = hipCreateChannelDesc(badBits, badBits, badBits, badBits, formatKind);
+
+  INFO("Number of bits: " << badBits);
+
+  const auto flag = GENERATE(from_range(std::begin(validFlags), std::end(validFlags)));
```

Comment:
I noticed that these lines of code are the same in a few negative tests. Moving them into a separate function which gets the hipChannelFormatKind as a param might prevent code repetition?
```cpp
  const auto flag = GENERATE(from_range(std::begin(validFlags), std::end(validFlags)));

  constexpr size_t s = 6;  // 6 to keep cubemap happy
  hipArray_t array{};

#if HT_NVIDIA
  hipError_t expectedError = hipErrorUnknown;
#else
  hipError_t expectedError = hipErrorInvalidValue;
#endif

  HIP_CHECK_ERROR(hipMalloc3DArray(&array, &desc, makeExtent(flag, s), flag), expectedError);
```

Reply:
nice, I've saved a bunch of lines